tests/suite/cli_misc.rs

```diff
@@ -713,6 +713,45 @@ info: toolchain 'test' uninstalled
     });
 }
 
+// issue #3737
+#[test]
+#[cfg(any(unix, windows))]
+fn toolchain_symlink() {
+    use rustup::utils::raw::symlink_dir;
+    use std::fs;
+
+    clitools::test(Scenario::SimpleV2, &|config| {
+        config.expect_ok(&["rustup", "toolchain", "install", "nightly"]);
```

Comment:
this installs a toolchain before running test code; I don't think you want/need that.

Reply:
I'm just repeating what's been reported in https://github.com/rust-lang/rustup/issues/3737#issue-2204264492: with a `RUSTUP_HOME` override and a symlink pointing to the default `toolchains` in it, Rustup should function correctly.

Of course I can do it another way around, which is creating a `toolchains` elsewhere and make a symlink under the default `RUSTUP_HOME`, install the nightly and verify it's there, uninstall it and verify it's gone.